### PR TITLE
Fix LLVM 14 related warnings.

### PR DESCRIPTION
**Pull request by xlauko**

### CODE REVIEW

File: include/remill/BC/Compat/PointerType.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "remill/BC/Version.h"
+
+namespace remill {
+
+inline static llvm::Type *getPointerElementType(llvm::PointerType *type) {
```
Comment (pgoodman) [inline include/remill/BC/Compat/PointerType.h:22]: Rename to `PointerElementType`. Is a checked needed for `type->isOpaque()` and have that return `nullptr`?

**Review by pgoodman — CHANGES_REQUESTED** [review]

File: include/remill/BC/Compat/PointerType.h (hunk 1 above)
Comment (xlauko) [reply include/remill/BC/Compat/PointerType.h:22]: Isn't it better to have the same name as LLVM does? So you don't have to always think whether it is with or without `get` prefix?

It is a precondition of both versions that the type is not opaque pointer. There is an assertion on that in both implementations.

Comment (pgoodman) [reply include/remill/BC/Compat/PointerType.h:22]: If you're putting it in the `llvm` namespace, use `get`. In the remill namespace, drop `get` or change to `Get`. Is it an assertion only enabled in debug builds, or all builds? If the former, make it return `nullptr`.

Comment (xlauko) [reply include/remill/BC/Compat/PointerType.h]: Fixed in c0b87c1ec6b4d2ff8d627a703ec663b4cf0f7ba8 and 9e87c962b3fccec6a4897341aa7c747e99ecf041

**Review by pgoodman — APPROVED** [review]

**Pull request merged**